packages/fxa-payments-server/src/components/PaymentConfirmation/index.test.tsx

```diff
@@ -229,6 +261,77 @@ describe('PaymentConfirmation', () => {
     expect(queryByText(defaultButtonLabel)).not.toBeInTheDocument();
   });
 
+  it('renders as expected with custom success button label text, from firestore config', () => {
+    updateConfig({
+      featureFlags: {
+        useFirestoreProductConfigs: true,
+      },
+    });
+    const subject = () => {
+      return render(
+        <PaymentConfirmation
+          {...{
+            profile: userProfile,
+            selectedPlan: selectedPlanWithConfiguration,
+            customer,
+            productUrl,
+          }}
+        />
+      );
+    };
+
+    const { queryByTestId, queryByText } = subject();
+    const subscriptionTitle = queryByTestId('subscription-success-title');
+    expect(subscriptionTitle).toBeInTheDocument();
```

Comment:
Ah yes it is, but it is the right way to do it. If it weren't in the doc, `subscriptionTitle` would be null if we use `queryByTestId`

We could replace `queryByTestId` with `getByTestId` which would throw an error if the ID isn't in the doc, so then we could remove the `toBeInTheDocument()` check.

Reply:
Ok, so this assertion is not about whether something from the product configuration object is rendered but just if _anything_ is rendered?  Otherwise I still cannot make the connection from the 'subscription-success-title' test id to something specific in the product configuration.

Reply:
@chenba better late than never, sorry for only getting to this now.

The primary objective of this test is to test if the button in `<PaymentConfirmation>` renders with the product configuration objects label. That's achieved by this part of the test, where after the page is rendered, the test queries to see if the custom label is on the rendered page. (And here we assume it's on the button)

```
expect(
      queryByText(
        selectedPlanWithConfiguration.configuration.uiContent
          .successActionButtonLabel
      )
    ).toBeInTheDocument();
    expect(queryByText(defaultButtonLabel)).not.toBeInTheDocument();
```

So technically we don't really need the other parts of the test, except for a way to maybe ensure the object is rendered correctly. (Since in this component`<SubscriptionTitle>` should always be rendered with test-id `subscription-success-title`)

I'm not sure if this answers your question though. Happy to keep talking about this though.

Reply:
Thanks for the detailed answer.  Got it.  👍